From 7d4b31e13d1aed82a5ce61752ed5e092f884e2ff Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Wed, 26 May 2021 15:43:25 +0300
Subject: [PATCH] [SW-0] greco: increase preboot size

PREBOOT size needs to be increased to accommodate new
code in preboot. Otherwise we get error while adding
some changes e.g.
with https://gerrit.habana-labs.com/#/c/97366/ following build
error observed,

-------------x-----------------
bin/preboot section `.stack' will not fit in region `.sram'
-------------x-----------------

Change-Id: I16d8a6b021bf5b76fd14aa64db0c9e904b2a4fc6
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/greco/common.h b/greco/common.h
index 7ee7ca9..091f731 100644
--- a/greco/common.h
+++ b/greco/common.h
@@ -11,7 +11,7 @@
 #include <bitops.h>
 
 #define PPBOOT_SIZE			0x5000		/* 20KB */
-#define PREBOOT_SIZE			0x10000		/* 64KB */
+#define PREBOOT_SIZE			0x10400		/* 65KB */
 
 #define SRAM_SIZE			0x8000000	/* 128MB */
 #define HL_SP_SRAM_SIZE			0x10000
diff --git a/greco_arc/common.h b/greco_arc/common.h
index 1eafa6e..611d37c 100644
--- a/greco_arc/common.h
+++ b/greco_arc/common.h
@@ -11,7 +11,7 @@
 #include <bitops.h>
 
 #define PPBOOT_SIZE		0x4000		/* 16KB */
-#define PREBOOT_SIZE		0x10000		/* 64KB */
+#define PREBOOT_SIZE		0x10400		/* 65KB */
 
 #define HL_FW_BOOT_PTRS_BASE		0x80000
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xC0000
diff --git a/greco_arc_gnu/common.h b/greco_arc_gnu/common.h
index af26eb2..d2d80ca 100644
--- a/greco_arc_gnu/common.h
+++ b/greco_arc_gnu/common.h
@@ -11,7 +11,7 @@
 #define SRAM_SIZE			0x8000000	/* 128MB */
 
 #define PPBOOT_SIZE			0x4000		/* 16KB */
-#define PREBOOT_SIZE			0x10000		/* 64KB */
+#define PREBOOT_SIZE			0x10400		/* 65KB */
 
 #define HL_FW_BOOT_PTRS_BASE		0x80000
 #define HL_SPI_LEGACY_PREBOOT_OFF	0xC0000
